lesson17/src/js/main.js

```diff
@@ -0,0 +1,225 @@
+  createTextBox.classList.add("");
+  createTextBox.textContent = error;
+  slideImageArea.appendChild(createTextBox);
+}
+
+function renderSlideArea() {
+  const slideImageArea = document.createElement("div");
+  slideImageArea.classList.add("mainvisual__images__inner");
+
+  const sliderArea = document.querySelector(".mainvisual__images");
+
+  sliderArea.appendChild(slideImageArea);
+}
+
+function renderSlideImage(item) {
+  const slideImageArea = document.querySelector(".mainvisual__images__inner");
+
+  const slider = document.createElement("ul");
+  slider.classList.add("slider");
+
+  const fragment = document.createDocumentFragment();
+  item.forEach((image, i) => {
+    const sliderItem = document.createElement("li");
+    sliderItem.classList.add("slider__item");
+    sliderItem.setAttribute("data-slide-index", i + 1);
+
+    const sliderImage = document.createElement("img");
+    sliderImage.classList.add("slider__image");
+    sliderImage.src = image.image;
+    sliderImage.width = image.width;
+    sliderImage.height = image.height;
+
+    if (i === 0) {
+      sliderItem.classList.add("is-active");
+    }
+
+    fragment.appendChild(sliderItem).appendChild(sliderImage);
+  });
+
+  slideImageArea.appendChild(slider).appendChild(fragment);
+}
+
+function renderSlideButton() {
+  let fragment = document.createDocumentFragment();
+
+  const buttonArea = document.createElement("div");
+  buttonArea.classList.add("slider__button");
+
+  const prevButton = document.createElement("button");
+  prevButton.setAttribute("class", "arrow prev");
+  prevButton.id = "js-button_prev";
+  prevButton.setAttribute("aria-label", "previous");
+  prevButton.textContent = "previous";
+
+  fragment.appendChild(prevButton);
+
+  const nextButton = document.createElement("button");
+  nextButton.setAttribute("class", "arrow next");
+  nextButton.id = "js-button_next";
+  nextButton.textContent = "next";
+  nextButton.setAttribute("aria-label", "next");
+
+  fragment.appendChild(nextButton);
+
+  const getSlideArea = document.querySelector(".mainvisual__images");
+
+  getSlideArea.appendChild(buttonArea).appendChild(fragment);
+}
+
+function renderPagenation(item) {
+  const pagenation = document.createElement("div");
+  pagenation.classList.add("slider__pagination");
+
+  let fragment = document.createDocumentFragment();
+
+  const current = document.createElement("span");
+  current.classList.add("current");
+  current.textContent = 1;
+  fragment.appendChild(current);
+
+  const separation = document.createElement("span");
+  separation.textContent = "/";
+  separation.classList.add("separation");
+  fragment.appendChild(separation);
+
+  const total = document.createElement("span");
+  total.classList.add("total");
+  total.textContent = item.length;
+  fragment.appendChild(total);
+
+  const getTextArea = document.querySelector(".mainvisual__textarea");
+  getTextArea.appendChild(pagenation).appendChild(fragment);
+}
+
+function pagenationCountDown() {
```

Comment:
IMO
I thought the function name should be a verb that describes what it does.
Same for the other 'pagenationCountUp' and 'ImageNextSlide' and ...more.

Reply:
I thought it was a CountDown verb, but there was no such verb here.
I reviewed the pagination function and corrected the function name.

[a04e25a](https://github.com/nyattsu72/hanson-frontend-cource/pull/25/commits/a04e25ab0cc6113d394fe52c759776830b782f3b)